app/frontend/wallet/shelley/transaction/computeTxPlan.ts

```diff
@@ -0,0 +1,327 @@
+import {InternalError, InternalErrorReason, UnexpectedErrorReason} from '../../../errors'
+import {CertificateType, Lovelace} from '../../../types'
+import {MAX_TX_OUTPUT_SIZE} from '../../constants'
+import {aggregateTokenBundles, getTokenBundlesDifference} from '../../helpers/tokenFormater'
+import {TxAuxiliaryData, TxCertificate, TxInput, TxOutput, TxWithdrawal} from '../../types'
+import {cborizeSingleTxOutput} from '../shelley-transaction'
+import {TxPlanResult} from './types'
+import {
+  createTokenChangeOutputs,
+  computeMinUTxOLovelaceAmount,
+  computeRequiredDeposit,
+  computeRequiredTxFee,
+} from './utils'
+import {encode} from 'borc'
+import {MAX_OUTPUT_TOKENS, MIN_UTXO_VALUE} from './constants'
+
+export function computeTxPlan(
+  inputs: Array<TxInput>,
+  outputs: Array<TxOutput>,
+  possibleChange: TxOutput,
+  certificates: Array<TxCertificate>,
+  withdrawals: Array<TxWithdrawal>,
+  auxiliaryData: TxAuxiliaryData
+): TxPlanResult {
+  const totalRewards = withdrawals.reduce((acc, {rewards}) => acc + rewards, 0)
+  const totalInput = inputs.reduce((acc, input) => acc + input.coins, 0) + totalRewards
+  const totalInputTokens = aggregateTokenBundles(inputs.map(({tokenBundle}) => tokenBundle))
+  const deposit = computeRequiredDeposit(certificates)
+  const totalOutput = outputs.reduce((acc, {coins}) => acc + coins, 0) + deposit
+  const totalOutputTokens = aggregateTokenBundles(outputs.map(({tokenBundle}) => tokenBundle))
+
+  // total amount of lovelace that had to be added to token-containing outputs
```

Comment:
perhaps a single bigger multiline comment at the beginning outlining the steps in human language here would help a lot in figuring out what's going on. The one-line comments scattered around the code help to some extent, but it's already too hard to get a bigger picture out of them - could you write such comment @PeterBenc ?